### PR TITLE
docs: add jsdocs and type definitions

### PR DESCRIPTION
**Pull request by snickbit**
I needed to write type definitions for this since my project is in TypeScript, so I figured I might as well take the time to add them directly to the package.

- Added basic JSDoc comments to all methods
- Included descriptions from the existing doc when possible
- Added typescript and tsc dev dependencies for type generation
- Ran linter & tests

### CODE REVIEW

**Comment by kleinron** [comment]
Hi @snickbit!
Great work, my fellow contributor!
A few comments:
1. At least on my machine, running the linter produced 3 error messages. You can run `lint-with-fix` to automatically fix these errors and push again.
2. Please increase the version in `package.json`. Since this is a significant change, I think version `0.3.0` is a good choice.
3. In `package.json`, the `types` element is pointing to the globally installed `tsc`. Please use the local (and thus portable) version installed in `./node_modules/.bin/tsc`. Eventually the line should look like: 
`"types": "./node_modules/.bin/tsc"`

**Comment by snickbit** [comment]
I updated it with your requests.

1. My IDE auto-formatted on commit the first time, so I missed those 3 spots. Whoops!
2. Bumped the version as requested
3. I updated the "types" script, but I used `node ./node_modules/typescript/bin/tsc` to maintain cross-platform compatibility, since `./node_modules/.bin/tsc` doesn't work on Windows. I hope that's okay!

File: types/util.d.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
+/**
+ * @return {void}
+ */
+export function bindMethods(): void;
+export class bindMethods {
+}
```
Comment (kleinron) [inline types/util.d.ts:6]: are you sure this is needed? maybe line #4 alone is enough?

**Comment by kleinron** [comment]
please also note that after resolving the linter issue, the relevant `.map` files should be regenerated and pushed, too.

File: types/util.d.ts (hunk 1 above)
Comment (snickbit) [reply types/util.d.ts:6]: Since you're binding methods to `this` it's considered a class constructor, so technically it's needed for TypeScript.

**Comment by snickbit** [comment]
Sorry about that. That's what I get for trying to do it quickly while at work.

**Pull request merged**